test/mocha/unit/rpc/FallbackHandler.test.ts

```diff
@@ -220,41 +80,220 @@ describe('FallbackHandler', () => {
+      )
+    })
+  })
 
-    await fallbackHandler.handler(alarmToInsufficientData)
-    expect(stubRepo.write.callCount).equals(0)
-    stubRepo.write.reset()
+  describe('verify we do DB update when OK', async () => {
+    let stubRepo: Sinon.SinonStubbedInstance<ProviderHealthStateDynamoDbRepository>
+    let alarmEvent: AlarmEvent
 
-    await fallbackHandler.handler(okToInsufficientData)
-    expect(stubRepo.write.callCount).equals(0)
-    stubRepo.write.reset()
+    beforeEach(() => {
+      stubRepo = Sinon.createStubInstance(ProviderHealthStateDynamoDbRepository)
+      stubRepo.read.resolves()
+      stubRepo.write.resolves()
+      stubRepo.update.resolves()
+      fallbackHandler['healthStateRepository'] = stubRepo
+
+      alarmEvent = {
+        alarmName: 'alarm1',
+        state: 'OK',
+        previousState: 'ALARM',
+        providerId: PROVIDER_ID,
+        reason: 'reason',
+      }
+    })
+
+    afterEach(() => {
+      stubRepo.read.reset()
+      stubRepo.write.reset()
+      stubRepo.update.reset()
+    })
+
+    it('no previous alarm, DB reads null', async () => {
+      stubRepo.read.resolves(null)
+      const healthinessUpdate = await fallbackHandler['updateDbItemWhenOk'](alarmEvent)
+
+      expect(stubRepo.update.callCount).equals(0)
+      expect(stubRepo.write.callCount).equals(0)
+      expect(healthinessUpdate).deep.equals(
+        new HealthinessUpdate(ProviderHealthiness.HEALTHY, ProviderHealthiness.HEALTHY)
+      )
+    })
+
+    it('no previous alarm, DB reads healthy provider state', async () => {
+      stubRepo.read.resolves({
+        healthiness: ProviderHealthiness.HEALTHY,
+        ongoingAlarms: [],
+        version: 1,
+      })
+      const healthinessUpdate = await fallbackHandler['updateDbItemWhenOk'](alarmEvent)
+
+      expect(stubRepo.update.callCount).equals(0)
+      expect(stubRepo.write.callCount).equals(0)
+      expect(healthinessUpdate).deep.equals(
+        new HealthinessUpdate(ProviderHealthiness.HEALTHY, ProviderHealthiness.HEALTHY)
+      )
+    })
+
+    it('has previous alarm, new alarm repeats', async () => {
+      stubRepo.read.resolves({
+        healthiness: ProviderHealthiness.UNHEALTHY,
+        ongoingAlarms: ['alarm1', 'alarm2'],
+        version: 1,
+      })
+      const healthinessUpdate = await fallbackHandler['updateDbItemWhenOk'](alarmEvent)
+
+      expect(stubRepo.write.callCount).equals(0)
+      expect(stubRepo.update.callCount).equals(1)
+      expect(stubRepo.update.getCall(0).args[0]).deep.equals(PROVIDER_ID)
+      expect(stubRepo.update.getCall(0).args[1]).deep.equals({
+        healthiness: ProviderHealthiness.UNHEALTHY,
+        ongoingAlarms: ['alarm2'],
+        version: 2,
+      })
+      expect(healthinessUpdate).deep.equals(
+        new HealthinessUpdate(ProviderHealthiness.UNHEALTHY, ProviderHealthiness.UNHEALTHY)
+      )
+    })
+
+    it('has previous alarm, new alarm repeats, all ongoing alarm cleared', async () => {
```

Comment:
if it repeats as an error state, it should stay on right? are you just assuming it can only flip/flop?

Reply:
This UT is for testing `updateDbItemWhenOk()` function, which means we received a `xxx` => `OK` event. The input event's `state` is `OK` as shown in line 210. It's not "new alarm repeats", it should be read as "new alarm event repeats". I've fixed the description.